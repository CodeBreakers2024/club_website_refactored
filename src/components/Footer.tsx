import React from 'react';
import Image from 'next/image';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCopyright } from '@fortawesome/free-regular-svg-icons';
import { faLinkedinIn, faInstagram } from '@fortawesome/free-brands-svg-icons';
import { faXTwitter as faTwitter } from '@fortawesome/free-brands-svg-icons';
import Link from 'next/link';

const Footer = () => {
  return (
    <footer className="h-[25vw] w-full mt-[30px] bg-[#1A1E1E] opacity-[0.97] relative bottom-0 left-0 right-0">
      <div className="h-[20vw] w-full bg-[#078888] absolute bottom-0 text-center">
        <p className="inline-block text-[8vw] font-[Franklin Gothic Heavy], 'Arial Narrow', Arial, sans-serif bg-gradient-to-b from-[#078888] to-[#1A1E1E] bg-clip-text text-transparent transform -translate-y-1/2">
          CODEBREAKERS
        </p>
        <p className="text-white text-[2vw] mt-[-6vw]">Seasoned. Nimble. Remote.</p>
        <hr className="mt-[2vw] text-white text-[1vw] mx-[12vw]" />
        <Image src={'/Assets/footer.svg'} alt="footer-img" className="absolute left-0 bottom-0 w-[40vw]" width={40} height={40} unoptimized />
        <div className="mt-[2vw] mx-[12vw] flex justify-around items-center text-[1.5vw] text-white font-light">
          <div className="flex opacity-[0.65]">
            <FontAwesomeIcon icon={faCopyright} />
            <p className="ml-[0.5vw] text-white text-[1.5vw]">Copyright Codebreakers</p>
          </div>
          <div>
            <ul className="list-none text-[1.5vw]">
              <li className="inline-block">
                <Link href="google.com" className="text-white pr-[1vw] opacity-[0.65]">Terms</Link>
              </li>
              <li className="inline-block">
                <Link href="google.com" className="text-white pr-[1vw] opacity-[0.65]">Privacy</Link>
              </li>
              <li className="inline-block">
                <Link href="google.com" className="text-white pr-[1vw] opacity-[0.65]">Cookies</Link>
              </li>
            </ul>
          </div>
          <div className="flex items-center space-x-[2vw]">
            <div className="w-[3vw] h-[3vw] border border-white bg-white rounded-full flex justify-center items-center">
              <Link
                href="https://www.linkedin.com/company/thecodebreakers-rcoem/mycompany/verification/"
                aria-label="LinkedIn"
                className="text-black text-[1.5vw]"
              >
                <FontAwesomeIcon icon={faLinkedinIn} />
              </Link>
            </div>
            <div className="w-[3vw] h-[3vw] border border-white bg-white rounded-full flex justify-center items-center">
              <Link
                href="https://www.instagram.com/thecodebreakers/"
                aria-label="Instagram"
                className="text-black text-[1.5vw]"
              >
                <FontAwesomeIcon icon={faInstagram} />
              </Link>
            </div>
            <div className="w-[3vw] h-[3vw] border border-white bg-white rounded-full flex justify-center items-center">
              <Link
                href="https://x.com/CodebreakersRBU"
                aria-label="Twitter"
                className="text-black text-[1.5vw]"
              >
                <FontAwesomeIcon icon={faTwitter} />
              </Link>
            </div>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
