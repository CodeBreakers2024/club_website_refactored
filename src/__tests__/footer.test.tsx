import { render, screen } from '@testing-library/react';
import Footer from '../components/Footer'; // Adjust the path according to your project structure
import '@testing-library/jest-dom';

describe('Footer Component', () => {
  it('should render the footer with the correct content', () => {
    render(<Footer />);

    // Check if "CODEBREAKERS" text is in the document
    expect(screen.getByText('CODEBREAKERS')).toBeInTheDocument();

    // Check if the "Seasoned. Nimble. Remote." text is in the document
    expect(screen.getByText('Seasoned. Nimble. Remote.')).toBeInTheDocument();

    // Check if the copyright statement is in the document
    expect(screen.getByText('Copyright Codebreakers')).toBeInTheDocument();

    // Check if the "Terms" link exists
    const termsLink = screen.getByText('Terms');
    expect(termsLink).toBeInTheDocument();
    expect(termsLink).toHaveAttribute('href', 'google.com');

    // Check if the "Privacy" link exists
    const privacyLink = screen.getByText('Privacy');
    expect(privacyLink).toBeInTheDocument();
    expect(privacyLink).toHaveAttribute('href', 'google.com');

    // Check if the "Cookies" link exists
    const cookiesLink = screen.getByText('Cookies');
    expect(cookiesLink).toBeInTheDocument();
    expect(cookiesLink).toHaveAttribute('href', 'google.com');

    // Check if the social media icons (LinkedIn, Instagram, Twitter) exist
    const linkedinIcon = screen.getByRole('link', { name: /LinkedIn/i });
    expect(linkedinIcon).toHaveAttribute('href', 'https://www.linkedin.com/company/thecodebreakers-rcoem/mycompany/verification/');

    const instagramIcon = screen.getByRole('link', { name: /Instagram/i });
    expect(instagramIcon).toHaveAttribute('href', 'https://www.instagram.com/thecodebreakers/');

    const twitterIcon = screen.getByRole('link', { name: /Twitter/i });
    expect(twitterIcon).toHaveAttribute('href', 'https://x.com/CodebreakersRBU');
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
