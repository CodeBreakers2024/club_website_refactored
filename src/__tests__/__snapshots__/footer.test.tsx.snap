// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="h-[25vw] w-full mt-[30px] bg-[#1A1E1E] opacity-[0.97] relative bottom-0 left-0 right-0"
  >
    <div
      class="h-[20vw] w-full bg-[#078888] absolute bottom-0 text-center"
    >
      <p
        class="inline-block text-[8vw] font-[Franklin Gothic Heavy], 'Arial Narrow', Arial, sans-serif bg-gradient-to-b from-[#078888] to-[#1A1E1E] bg-clip-text text-transparent transform -translate-y-1/2"
      >
        CODEBREAKERS
      </p>
      <p
        class="text-white text-[2vw] mt-[-6vw]"
      >
        Seasoned. Nimble. Remote.
      </p>
      <hr
        class="mt-[2vw] text-white text-[1vw] mx-[12vw]"
      />
      <img
        alt="footer-img"
        class="absolute left-0 bottom-0 w-[40vw]"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/Assets/footer.svg"
        style="color: transparent;"
        width="40"
      />
      <div
        class="mt-[2vw] mx-[12vw] flex justify-around items-center text-[1.5vw] text-white font-light"
      >
        <div
          class="flex opacity-[0.65]"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-copyright "
            data-icon="copyright"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 48a208 208 0 1 1 0 416 208 208 0 1 1 0-416zm0 464A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM199.4 312.6c-31.2-31.2-31.2-81.9 0-113.1s81.9-31.2 113.1 0c9.4 9.4 24.6 9.4 33.9 0s9.4-24.6 0-33.9c-50-50-131-50-181 0s-50 131 0 181s131 50 181 0c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0c-31.2 31.2-81.9 31.2-113.1 0z"
              fill="currentColor"
            />
          </svg>
          <p
            class="ml-[0.5vw] text-white text-[1.5vw]"
          >
            Copyright Codebreakers
          </p>
        </div>
        <div>
          <ul
            class="list-none text-[1.5vw]"
          >
            <li
              class="inline-block"
            >
              <a
                class="text-white pr-[1vw] opacity-[0.65]"
                href="google.com"
              >
                Terms
              </a>
            </li>
            <li
              class="inline-block"
            >
              <a
                class="text-white pr-[1vw] opacity-[0.65]"
                href="google.com"
              >
                Privacy
              </a>
            </li>
            <li
              class="inline-block"
            >
              <a
                class="text-white pr-[1vw] opacity-[0.65]"
                href="google.com"
              >
                Cookies
              </a>
            </li>
          </ul>
        </div>
        <div
          class="flex items-center space-x-[2vw]"
        >
          <div
            class="w-[3vw] h-[3vw] border border-white bg-white rounded-full flex justify-center items-center"
          >
            <a
              aria-label="LinkedIn"
              class="text-black text-[1.5vw]"
              href="https://www.linkedin.com/company/thecodebreakers-rcoem/mycompany/verification/"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-linkedin-in "
                data-icon="linkedin-in"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
          <div
            class="w-[3vw] h-[3vw] border border-white bg-white rounded-full flex justify-center items-center"
          >
            <a
              aria-label="Instagram"
              class="text-black text-[1.5vw]"
              href="https://www.instagram.com/thecodebreakers/"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-instagram "
                data-icon="instagram"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
          <div
            class="w-[3vw] h-[3vw] border border-white bg-white rounded-full flex justify-center items-center"
          >
            <a
              aria-label="Twitter"
              class="text-black text-[1.5vw]"
              href="https://x.com/CodebreakersRBU"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-x-twitter "
                data-icon="x-twitter"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
